upgrade-2.x.sh

```diff
@@ -215,6 +215,19 @@ function _run_supervisor_update() {
     return "${ret}"
 }
 
+function _fetch_supervisor_version() {
+    local resp
+    local supervisor_version
+
+    resp=$(CURL_CA_BUNDLE=${TMPCRT} curl --silent --retry 10 --header "Authorization: Bearer ${APIKEY}" "${API_ENDPOINT}/v6/device(uuid='${UUID}')?\$select=supervisor_version")
```

Comment:
Heads-up that the check in the API compares the current `device.should_be_managed_by__supervisor_release` (not the `device.supervisor_version`, even though someone could argue we should test both) and compares it with the new targer supervisor release.
See: https://github.com/balena-io/open-balena-api/blob/d632e075e543bd7c8125fe97559631efbcb56831/src/features/supervisor-app/hooks/supervisor-app.ts#L101-L131

Reply:
thanks @thgreasi I pushed a new version that checks `should_be_managed_by__supervisor_release` and aborts if its newer than the `supervisor_version` as we don't want two simultaneous updates going on.